controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -73,6 +73,106 @@ static bool validate_hex_notation(const std::string &str, uint8_t expected_octet
     return true;
 }
 
+/**
+ * @brief Parse bss_info string into bss_info_conf_struct.
+ * 
+ * @param[in] bss_info_str String conataining bss info configuration
+ * @param[out] bss_info_conf Controller database struct filled with configuration.
+ * @param[out] err_string Contains an error description if the function fails.
+ * @return true if successful, false if not.
+ */
+static std::string parse_bss_info(const std::string &bss_info_str,
+                                  db::bss_info_conf_t &bss_info_conf, std::string &err_string)
+{
+    auto confs = string_utils::str_split(bss_info_str, ' ');
+    if (confs.size() != 8) {
+        err_string = "missing configuration";
+        return std::string();
+    }
+
+    // Alid
+    std::string al_mac(confs[0]);
+    if (!net::network_utils::is_valid_mac(al_mac)) {
```

Comment:
Actually, IMO it would be better to use sMacAddr in the database (*), and therefore also use mac_from_string here.

But now I notice that that function doesn't even have proper error handling :-(

(*) Yes, I realize that the database currently uses strings all over the place to store MAC addresses, but I think that that is wrong and that we should (gradually) change it.

Reply:
I guess it is easier to work with string...
But maybe you are right, anyway, maybe we will do it as part of database refactoring.
(There a lot of things to refactor there...)

Reply:
So can you open a task for this refactoring?

Reply:
Well, it will require DR first.

Reply:
No need for a full DR to open a refactoring task: #243